core/unix/os.c

```diff
@@ -6418,7 +6418,6 @@ set_stdfile_fileno(stdfile_t **stdfile, file_t old_fd, file_t file_no)
     (*stdfile)->STDFILE_FILENO = file_no;
 #    endif
 #else
-#    warning stdfile_t is opaque; DynamoRIO will not set fds of libc FILEs.
```

Comment:
We try to separate the build system from the target runtime system: but here it seems they are tied together so a glibc build will fail when trying to use a musl libc library with a tool run on a musl system; and this musl build will fail on a glibc system even though the musl build would be able to work if it had dynamic discovery of the libc type.  Add a XXX comment on that possibility?  Likely not worth effort to implement at this point.

Reply:
Sounds good.  Maybe add "instead of using build-time preprocessor defines" to make it even more clear.